File: yeongseong/5week/예산.js
function solution(d, budget) {
    var answer = 0;
    // 오름차순 정렬
    d.sort((a, b) => a - b);
    for (let i = 0; i < d.length; i++){
        // 예산이 현재 요소보다 크거나 같은 경우
        if (budget >= d[i]){
            // 현재 요소를 예산에서 빼기
            budget -= d[i];
            // 결과 카운터 증가
            answer++
        }else {
            break;
        }
    }
    return answer;
}

File: dongmin/week5/예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;

        Arrays.sort(d);     //정렬을 통해 낮은 값 부터 계산하며 최대한 많은 부서의 신청을
                            //받을 수 있도록 한다.

        for (int i = 0; i < d.length; i++) {
            if(budget-d[i]>=0){         //예산에서 신청 금액을 뺀 값이 0이상 일 경우
                budget-=d[i];
                answer++;
            }
        }
        return answer;
    }
}